src/rdkafka_event.c

```diff
@@ -224,6 +224,37 @@ int rd_kafka_event_log (rd_kafka_event_t *rkev, const char **fac,
 	return 0;
 }
 
+int rd_kafka_event_debug_contexts (rd_kafka_event_t *rkev,
+            char *dst, size_t dstsize) {
+        static const char *names[] = {
+                "generic",
+                "broker",
+                "topic",
+                "metadata",
+                "feature",
+                "queue",
+                "msg",
+                "protocol",
+                "cgrp",
+                "security",
+                "fetch",
+                "interceptor",
+                "plugin",
+                "consumer",
+                "admin",
+                "eos",
+                "mock",
+                NULL
+        };
+        int ctx = rkev->rko_u.log.ctx;
```

Comment:
This is not safe to do prior to checking the type.
Just use `rkev->rko_u.log.ctx` directly when calling flags2str.